From baf854a7f8c7ac573f06310b0e994b9f56b9ade5 Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Mon, 30 Mar 2015 10:43:34 +0200
Subject: [PATCH 0428/1800] EHTMLEditorView - Newly inserted tables have nearly
 zero height

When inserting a new table into contenteditable element the width of the cells
is nearly zero and the td { min-height } doesn't work so put unicode zero width
space before each cell.
---
 e-util/e-html-editor-view.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/e-util/e-html-editor-view.c b/e-util/e-html-editor-view.c
index 5433b795ec..14f14dfb0e 100644
--- a/e-util/e-html-editor-view.c
+++ b/e-util/e-html-editor-view.c
@@ -2493,6 +2493,15 @@ e_html_editor_view_update_fonts (EHTMLEditorView *view)
 		"  margin-bottom: 0.2em;\n"
 		"}\n");
 
+	/* When inserting a table into contenteditable element the width of the
+	 * cells is nearly zero and the td { min-height } doesn't work so put
+	 * unicode zero width space before each cell. */
+	g_string_append (
+		stylesheet,
+		"td::before {\n"
+		" content: \"\xe2\x80\x8b\";"
+		"}\n");
+
 	g_string_append (
 		stylesheet,
 		"img "
-- 
2.15.0

